File: javaLoginCore/src/main/java/com/hamdan/forzenbook/java/login/core/data/database/LoginEntity.java
package com.hamdan.forzenbook.java.login.core.data.database;

import static com.hamdan.forzenbook.java.login.core.data.database.LoginEntity.TABLE_NAME;

import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

@Entity(tableName = TABLE_NAME)
public class LoginEntity {

    final static String TABLE_NAME = "login";
    final static String DATABASE_ID = "login_database_id";
    final static String TOKEN = "login_token";
    final static String EMAIL = "login_email";

    @ColumnInfo(name = TOKEN)
    public String token;

    @ColumnInfo(name = EMAIL)
    public String email;

    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = DATABASE_ID)
    public int loginId = 0;
}

File: loginCore/src/main/java/com/hamdan/forzenbook/login/core/data/database/LoginEntity.kt
package com.hamdan.forzenbook.login.core.data.database

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey
import com.hamdan.forzenbook.login.core.data.database.LoginEntity.Companion.TABLE_NAME

@Entity(tableName = TABLE_NAME)
data class LoginEntity(
    @ColumnInfo(name = TOKEN) val token: String,
    @ColumnInfo(name = EMAIL) val email: String,
) {
    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = DATABASE_ID)
    var loginId: Int = 0

    companion object {
        const val TABLE_NAME = "login"
        const val DATABASE_ID = "login_database_id"
        const val TOKEN = "login_token"
        const val EMAIL = "login_email"
    }
}
